From 4ef3cc99343836690bdcc56de94b9587a326a83e Mon Sep 17 00:00:00 2001
From: "Sharma, AmitX A" <amitx.a.sharma@intel.com>
Date: Wed, 18 Nov 2020 21:44:27 +0530
Subject: [PATCH] UGW_SW-54855: Added a new ioctl command
 (PPA_CMD_GET_BRIDGE_FID) for getting bridge Fid parameter from ppacmd tool

---
 include/uapi/net/ppa_api.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/include/uapi/net/ppa_api.h b/include/uapi/net/ppa_api.h
index 5fdad78baf82..1fef6f6f3ca9 100755
--- a/include/uapi/net/ppa_api.h
+++ b/include/uapi/net/ppa_api.h
@@ -1139,6 +1139,14 @@ typedef struct {
 	uint32_t bridge_enable; /*!< enable/disable bridging mac address learning flag */
 	uint32_t flags; /*!< reserved for future */
 } PPA_CMD_BRIDGE_ENABLE_INFO;
+/*!
+	\brief This is the data structure to get the Fid id For IOCTL.
+ */
+typedef struct {
+	PPA_IFNAME ifname[PPA_IF_NAME_SIZE]; /*!< the Bridge interface name */
+	uint16_t fid; /*!< the Fid */
+	uint32_t flags; /*!< reserved for future */
+} PPA_CMD_BRIDGE_FID_INFO;
 /*!
 	\brief QoS Shaper Mode Configuration
  */
@@ -2050,6 +2058,7 @@ typedef union {
 	PPA_CMD_DEL_SESSION_INFO del_session; /*!< PPA session delete info. */
 	PPA_QOS_METER_INFO meter_info; /*!< PPA qos meter info. */
 	PPA_QOS_METER_CFG meter_cfg; /*!< PPA qos meter configuration. */
+	PPA_CMD_BRIDGE_FID_INFO br_fid_info; /*!< PPA bridge fid parameter */
 } PPA_CMD_DATA;
 /*@}*/ /* PPA_IOCTL */
 /* -------------------------------------------------------------------------- */
@@ -2173,6 +2182,7 @@ typedef enum {
 	PPA_CMD_QOS_DSCP_CLASS_RESET_NR, /*!< NR for PPA_CMD_QOS_DSCP_CLASS_RESET */
 	PPA_CMD_GET_IFACE_MIB_NR, /*!< NR for PPA_CMD_GET_IFACE_MIB */
 	PPA_CMD_ADD_QOS_METER_NR, /*!< NR for PPA_CMD_SET_QOS_METER */
+	PPA_CMD_GET_BRIDGE_FID_NR, /*!< NR for PPA_CMD_GET_BRIDGE_FID */
 	/* PPA_IOC_MAXNR should be the last one in the enumberation */
 	PPA_IOC_MAXNR /*!< NR for PPA_IOC_MAXNR */
 } PPA_IOC_NR;
@@ -2440,6 +2450,11 @@ typedef enum {
 	\ref PPA_CMD_GET_BRIDGE_STATUS structure
  */
 #define PPA_CMD_GET_BRIDGE_STATUS _IOR(PPA_IOC_MAGIC, PPA_CMD_GET_BRIDGE_STATUS_NR, PPA_CMD_BRIDGE_ENABLE_INFO)
+/** PPA get ppa bridge fid. Value is manipulated by _IOR() macro for final value
+	\param PPA_CMD_GET_BRIDGE_FID The parameter points to a
+	\ref PPA_CMD_GET_BRIDGE_FID structure
+ */
+#define PPA_CMD_GET_BRIDGE_FID _IOR(PPA_IOC_MAGIC, PPA_CMD_GET_BRIDGE_FID_NR, PPA_CMD_BRIDGE_FID_INFO)
 /* Classification start*/
 /** PPA Add QOS Class. Value is manipulated by _IOW() macro for final value
 	\param[in,out] PPA_CMD_CLASSIFIER_INFO The parameter points to a
